## 1. 상관관계분석

In [ ]:
df.corr()

## 2. 정규성검정

### 일반 정규성검정

In [10]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/normal1.csv')

df

data
0    295.979767
1    289.703739
2    300.609670
3    299.890856
4    293.159612
..          ...
995  299.901120
996  307.387808
997  303.869390
998  313.979589
999  306.950007

[1000 rows x 1 columns]

In [11]:
stats.shapiro(df)

ShapiroResult(statistic=0.9981446862220764, pvalue=0.3485966622829437)

- H0: 데이터가 정규분포를 따른다.
- p-value가 0.05보다 크므로 유의수준 5%에서 귀무가설을 기각할 수 없다. 따라서 데이터가 정규분포를 따른다고 결론내릴 수 있다.

### 로그변환 후 정규성 검정

In [12]:
import pandas as pd
import numpy as np
import scipy.stats as stats

df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/normal3.csv')

df

data
0    0.880145
1   -0.021619
2    0.509888
3    0.179442
4    0.493778
..        ...
995 -0.229930
996  0.087762
997  0.651349
998  2.489877
999  0.704755

[1000 rows x 1 columns]

In [13]:
log_df = np.log1p(df)

stats.shapiro(log_df)

ShapiroResult(statistic=0.9976893067359924, pvalue=0.17551212012767792)

### 데이터가 5000개 이상일 때 -> anderson

In [16]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/normal6.csv')
df

data
0     296.016803
1     299.628476
2     298.232057
3     293.256749
4     301.037710
...          ...
5995  304.008424
5996  301.043219
5997  302.542420
5998  294.456145
5999  301.733819

[6000 rows x 1 columns]

In [19]:
df.data.values

array([296.01680341, 299.62847649, 298.23205699, ..., 302.54242027,
       294.45614507, 301.73381948])

In [20]:
stats.anderson(df.data.values)

AndersonResult(statistic=0.8266993530414766, critical_values=array([0.576, 0.656, 0.786, 0.917, 1.091]), significance_level=array([15. , 10. ,  5. ,  2.5,  1. ]))

p-value 0.05 -> 0.786  
p-value 0.025 -> 0.917  
따라서 p-value가 0.05보다 작으므로 귀무가설을 기각한다. 결과적으로 데이터는 정규성을 가진다고 볼 수 없다.

## 3. 일원배치분산분석

## 4. 이원배치분산분석

## 5. 단일표본 t검정

100명의 키 정보가 들어있는 데이터가 있다. 데이터가 정규성을 만족하는지 확인하라. 그리고 평균 키는 165라고 판단할 수 있는지 귀무가설과 대립가설을 설정한 후, 유의수준 5%에서 검정하라.

### 모수적 검정

In [ ]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/height1.csv')

stats.shapiro(df)   # 정규성 가짐

ShapiroResult(statistic=0.9872670769691467, pvalue=0.455832302570343)

In [24]:
df.head()

height
0  160.237691
1  164.747324
2  165.401628
3  168.801627
4  153.199021

In [25]:
stats.ttest_1samp(df.height, 165)

Ttest_1sampResult(statistic=3.2017884987150644, pvalue=0.0018367171548080209)

- H0: 100명의 키의 평균은 165이다.
- H1: 100명의 키의 평균은 165가 아니다.

정규성 검정 결과 p-value가 0.05 이상이므로 귀무가설을 기각할 수 없어 데이터는 정규성을 가진다고 할 수 있다.
단일표본 t검정 결과 p-value가 0.05 미만이므로 귀무가설을 기각하여 100명의 키의 평균은 165가 아니라고 결론내릴 수 있다.

### 비모수적 검정

In [ ]:
import pandas as pd
import scipy.stats as stats
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/height2.csv')

stats.shapiro(df)   # 정규성 X

ShapiroResult(statistic=0.9672006368637085, pvalue=0.013552471995353699)

In [ ]:
stats.wilcoxon(df.height)   # 암튼 기각인데 어캐하는지는 잘..

WilcoxonResult(statistic=0.0, pvalue=3.87726171873134e-18)

## 6. 독립표본 t검정

## 7. 점추정과 구간추정

## 8. 베르누이/이항분포

## 9. 카이제곱 독립성 검정

## 10. 포아송 분포

## 11. 지지도 / 향상도 / 신뢰도

## 12. 카이제곱 적합성 검정

## 13. 로지스틱 회귀분석

In [1]:
import statsmodels.api as sm

## 14. 선형회귀